src/main/java/org/broadinstitute/hellbender/tools/spark/sv/Kmer.java

```diff
@@ -0,0 +1,177 @@
+    {
+        final long mask = (1L << kSize) - 1L;
+        final long newV1 = ((v1 << 2) | (v2 >> (kSize-2))) & mask;
+        final long newV2 = ((v2 << 2) | (base & 3L)) & mask;
+        return new Kmer(newV1,newV2);
+    }
+
+    /**
+     * Add a new leading base to a Kmer, discarding the trailing base.
+     * E.g., if kkk.toString(5) is "ACTGA", then kkk.predecessor(3,5).toString(5) is "TACTG".
+     * @param base must be 0, 1, 2, or 3, corresponding to A, C, G, or T.
+     */
+    public Kmer predecessor( final long base, final int kSize )
+    {
+        final long mask = (1L << kSize) - 1L;
+        final long newV1 = ((v1 >> 2) | (base << (kSize-2))) & mask;
+        final long newV2 = ((v2 >> 2) | (v1 << (kSize-2))) & mask;
+        return new Kmer(newV1,newV2);
+    }
+
+    /**
+     * Returns a new Kmer that's the reverse-complement of this one.
+     * E.g., if kkk.toString(5) is "ACTGA", then kkk.rc(5).toString(5) is "TCAGT".
+     */
+    public Kmer rc( final int kSize )
+    {
+        final long mask = (1L << kSize) - 1L;
+        int compK = 64 - kSize;
+        if ( (kSize&1) == 0 )
+            return new Kmer(rcLong(v2 << compK)&mask,rcLong(v1 << compK)&mask);
+        long newV1 = ((rcLong(v2 << (compK+1)) << 1) & mask) ^ ((v1 & 1L) ^ 1L);
+        long newV2 = (rcLong(v1 << compK) & mask) ^ (v2 & (1L << (kSize-1)));
+        return new Kmer(newV1, newV2);
+    }
+
+    /**
+     * Returns a Kmer that is a canonical representation of this one.
+     * An odd-K Kmer is in canonical form if its middle base is A or C.
+     * The reverse-complement of a non-canonical Kmer is a canonical Kmer, and vice versa.  (Think about it.)
+     * Canonical form is not defined for even-K Kmers (too expensive to compute routinely).
+     */
+    public Kmer canonical( final int kSize )
+    {
+        if ( (kSize & 1) == 0 ) throw new IllegalArgumentException("K must be odd to canonicalize.");
+        if ( (v1 & 1L) == 0 ) return this;
+        return rc(kSize);
+    }
+
+    @Override
+    public boolean equals( final Object obj )
+    {
+        if ( !(obj instanceof Kmer) ) return false;
+        final Kmer that = (Kmer)obj;
+        return this.v1 == that.v1 && this.v2 == that.v2;
+    }
+
+    public boolean equals( final Kmer that )
+    {
+        return this.v1 == that.v1 && this.v2 == that.v2;
+    }
+
+    @Override
+    public int hashCode() { return (int)(v1 ^ (v1 >> 32) ^ v2 ^ (v2 >> 32)); }
+
+    /**
+     * Kmer comparison is consistent with equals.
+     * It's also the same as the lexicographic ordering you'd get using toString on the Kmers.
+     */
+    @Override
+    public int compareTo( final Kmer that )
+    {
+        int result = Long.compare(this.v1,that.v1);
+        if ( result == 0 ) result = Long.compare(this.v2,that.v2);
+        return result;
+    }
+
+    /**
+     * Not an override.  A Kmer doesn't know what K is, so it has to be supplied.
+     */
+    public String toString( final int kSize )
```

Comment:
Valid?  Not really.  It won't explode or anything, but it won't be particularly meaningful.  You're expected to track K externally.
It would be far more straightforward if a Kmer knew its size, for sure.  The use case is that you have a compile-time constant that specifies K for the purposes of some program, and a zillion instances of Kmers of that size that are getting shuffled around by Spark pipelines.  It's too costly to store K when it's effectively a constant, anyway.  We can write another wrapper class that has as fields one of these plus a K value if we ever find a need to.  That class can implement the methods with the K-less signatures that we'd prefer to see.